feature/metis/conversation/src/main/kotlin/de/tum/informatics/www1/artemis/native_app/feature/metis/conversation/ui/post/post_actions/PostActionBar.kt

```diff
@@ -0,0 +1,141 @@
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.pluralStringResource
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.unit.dp
+import de.tum.informatics.www1.artemis.native_app.core.ui.material.colors.PostColors
+import de.tum.informatics.www1.artemis.native_app.feature.metis.conversation.R
+import de.tum.informatics.www1.artemis.native_app.feature.metis.shared.content.dto.DisplayPriority
+import de.tum.informatics.www1.artemis.native_app.feature.metis.shared.content.dto.IStandalonePost
+
+internal const val TEST_TAG_PIN_POST = "TEST_TAG_PIN_POST"
+internal const val TEST_TAG_POST_EDIT = "TEST_TAG_POST_EDIT"
+internal const val  TEST_TAG_POST_DELETE = "TEST_TAG_POST_DELETE"
+
+
+@Composable
+fun PostActionBar(
+    modifier: Modifier,
+    post: IStandalonePost,
+    postActions: PostActions,
+    repliesCount: Int
+) {
+    HorizontalDivider()
+
+    Row(
+        modifier = modifier.height(28.dp),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        Text(
+            modifier = Modifier.padding(horizontal = 8.dp),
+            text = pluralStringResource(
+                id = R.plurals.post_replies_count,
+                count = repliesCount,
+                repliesCount
+            ),
+            color = MaterialTheme.colorScheme.secondary,
+            style = MaterialTheme.typography.bodyMedium
+        )
+
+        Spacer(modifier = Modifier.weight(1f))
+
+        ActionBar(post, postActions)
+    }
+
+    HorizontalDivider()
+}
+
+@Composable
+private fun ActionBar(
+    post: IStandalonePost,
+    postActions: PostActions
+) {
+    Row(
+        modifier = Modifier.horizontalScroll(rememberScrollState()),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        IconButton(
+            onClick = {
+                postActions.onCopyText()
+            }
+        ) {
+            Icon(
+                imageVector = Icons.Default.ContentCopy,
+                tint = MaterialTheme.colorScheme.primary,
+                contentDescription = null
+            )
+        }
+
+        postActions.onPinPost?.let {
+            VerticalDivider()
+
+            IconButton(
+                modifier = Modifier.testTag(TEST_TAG_PIN_POST),
+                onClick = {
+                    postActions.onPinPost.invoke()
+                }
+            ) {
+                if (post.displayPriority == DisplayPriority.PINNED) Icon(
+                    ImageVector.vectorResource(
+                        R.drawable.unpin
+                    ), null, tint = MaterialTheme.colorScheme.primary
+                ) else Icon(ImageVector.vectorResource(R.drawable.pin), null, tint = MaterialTheme.colorScheme.primary)
+            }
+        }
+
+        if (postActions.requestEditPost != null || postActions.requestDeletePost != null) {
+            VerticalDivider()
+        }
```

Comment:
I'm a big fan of return early, we could use this here :D
```
if (postActions.requestEditPost == null && postActions.requestDeletePost == null) {
    return
}

VerticalDivider()
```

Reply:
True :)